sn_peers_acquisition/src/lib.rs

```diff
@@ -124,14 +124,20 @@ async fn get_network_contacts(args: &PeersArgs) -> Result<Vec<Multiaddr>> {
 pub fn parse_peer_addr(addr: &str) -> Result<Multiaddr> {
     // Parse valid IPv4 socket address, e.g. `1.2.3.4:1234`.
     if let Ok(addr) = addr.parse::<std::net::SocketAddrV4>() {
-        #[cfg(not(feature = "quic"))]
+        let multiaddr = Multiaddr::from(*addr.ip());
+        #[cfg(feature = "tcp")]
         // Turn the address into a `/ip4/<ip>/tcp/<port>` multiaddr.
-        let multiaddr = Multiaddr::from(*addr.ip()).with(Protocol::Tcp(addr.port()));
+        let multiaddr = multiaddr.with(Protocol::Tcp(addr.port()));
         #[cfg(feature = "quic")]
         // Turn the address into a `/ip4/<ip>/udp/<port>/quic-v1` multiaddr.
-        let multiaddr = Multiaddr::from(*addr.ip())
+        let multiaddr = multiaddr
             .with(Protocol::Udp(addr.port()))
             .with(Protocol::QuicV1);
+        #[cfg(target_arch = "wasm32")]
```

Comment:
Is it possbile to enable more than one of these features here? This might lead to unexpected behaviours. Can we somehow have a check for use of multiple incompatible features?

Reply:
We'll add some compile checks etc in down the line. I'll be following this up 👍